packages/datepickers/src/styled/StyledMenu.tsx

```diff
@@ -64,11 +64,11 @@ interface IStyledMenuViewProps extends HTMLProps<HTMLUListElement> {
  * Accepts all `<ul>` props
  */
 const StyledMenuView = styled.ul.attrs<IStyledMenuViewProps>(props => ({
-  'data-garden-id': COMPONENT_ID,
+  'data-garden-id': (props as any['data-garden-id']) || COMPONENT_ID,
```

Comment:
Did you determine this "or" treatment is needed after all?

Reply:
This isn't needed.